features/step_definitions/keygen_steps.rb

```diff
@@ -13,9 +14,10 @@
   address = ROOT_ADDRESS.split(':').slice(0,2).join(':') + ":999"
   client = BitPay::SDK::Client.new(api_uri: address, pem: pem, insecure: true)
   begin
+    sleep 1 # rate limit compliance
     client.pair_pos_client("1ab2c34")
     raise "pairing unexpectedly worked"
-  rescue => error
+  rescue BitPay::ConnectionError => error
```

Comment:
Naming the error isn't needed here and is in fact (I think) counterproductive. We want the test to fail and the test suite to continue without throwing an error, we want it to return the error type as a response. If the rescue hits the wrong error, cucumber will pick that up when comparing the error expected with the error received.